qa/rpc-tests/wallet_nullifiers.py

```diff
@@ -0,0 +1,162 @@
+                break
+
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # add zaddr to node 2
+        myzaddr = self.nodes[2].z_getnewaddress()
+
+        # import node 2 zaddr into node 1
+        myzkey = self.nodes[2].z_exportkey(myzaddr)
+        self.nodes[1].z_importkey(myzkey)
+
+        # encrypt node 1 wallet and wait to terminate
+        self.nodes[1].encryptwallet("test")
+        bitcoind_processes[1].wait()
+
+        # restart node 1
+        self.nodes[1] = start_node(1, self.options.tmpdir)
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 1, 2)
+        self.sync_all()
+
+        # send node 0 zaddr to note 2 zaddr
+        recipients = []
+        recipients.append({"address":myzaddr, "amount":7.0})
+        myopid = self.nodes[0].z_sendmany(myzaddr0, recipients)
+
+        opids = []
+        opids.append(myopid)
+
+        timeout = 120
+        status = None
+        for x in xrange(1, timeout):
+            results = self.nodes[0].z_getoperationresult(opids)
+            if len(results)==0:
+                sleep(1)
+            else:
+                status = results[0]["status"]
+                assert_equal("success", status)
+                mytxid = results[0]["result"]["txid"]
+                break
+
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # check zaddr balance
+        zsendmanynotevalue = Decimal('7.0')
+        assert_equal(self.nodes[2].z_getbalance(myzaddr), zsendmanynotevalue)
+        assert_equal(self.nodes[1].z_getbalance(myzaddr), zsendmanynotevalue)
+
+        # add zaddr to node 3
+        myzaddr3 = self.nodes[3].z_getnewaddress()
+
+        # send node 2 zaddr to note 3 zaddr
+        recipients = []
+        recipients.append({"address":myzaddr3, "amount":2.0})
+        myopid = self.nodes[2].z_sendmany(myzaddr, recipients)
+
+        opids = []
+        opids.append(myopid)
+
+        timeout = 120
+        status = None
+        for x in xrange(1, timeout):
+            results = self.nodes[2].z_getoperationresult(opids)
+            if len(results)==0:
+                sleep(1)
+            else:
+                status = results[0]["status"]
+                assert_equal("success", status)
+                mytxid = results[0]["result"]["txid"]
+                break
+
+        self.nodes[2].generate(1)
+        self.sync_all()
+
+        # check zaddr balance
+        zsendmany2notevalue = Decimal('2.0')
+        zsendmanyfee = Decimal('0.0001')
+        zaddrremaining = zsendmanynotevalue - zsendmany2notevalue - zsendmanyfee
+        assert_equal(self.nodes[3].z_getbalance(myzaddr3), zsendmany2notevalue)
+        assert_equal(self.nodes[2].z_getbalance(myzaddr), zaddrremaining)
+
+        # Parallel encrypted wallet can't cache nullifiers for received notes,
+        # and therefore can't detect spends. So it sees a balance corresponding
+        # to the sum of both notes it received (one as change).
```

Comment:
Ouch, I hadn't realised that the balance seen due to this problem could be greater than the amount the wallet ever had, due to change. ref #1528 . (Does not block.)